### PR TITLE
Add some security considerations text

### PR DESCRIPTION
**Pull request by chris-wood**

### CODE REVIEW

File: draft-ietf-tls-external-psk-importer.md [hunk 1]
```diff
@@ -195,7 +203,33 @@ PSKs for TLS 1.2. Indeed, this is necessary for incremental deployment.
...
+protocols, wherein at least one is uncompromised, jointly authenticates all protocols.
+Authenticating with an externally provisioned PSK, therefore, should ideally authenticate both
+the TLS connection and the external provision process. Typically, the external provision process
+produces a PSK and corresponding context in which the PSK should be used. We refer to an external
+PSK without such context as "context free".
+
+Thus, in considering the source-independence and compound authentication requirements, the Key Import
+API described in this document aims to achieve the following goals:
+
+1. Externally provisioned PSKs imported into TLS achieve compound authentication of the provision step and connection.
```
Comment (jhoyla) [inline draft-ietf-tls-external-psk-importer.md]: `s/step/steps/`

File: draft-ietf-tls-external-psk-importer.md [hunk 2]
```diff
@@ -195,7 +203,33 @@ PSKs for TLS 1.2. Indeed, this is necessary for incremental deployment.
...
+assume that input keying material (IKM) is sampled from some "source" probability distribution
+and that any two IKM values are chosen independently of each other {{Kraw10}}. This
+source-independence requirement implies that the same IKM value cannot be used for two different
+KDFs.
+
+PSK-based authentication is functionally equivalent to session resumption in that a connection
+uses existing key material to authenticate both endpoints. Following the work of
+{{?BAA15=DOI.10.14722/ndss.2015.23277}}, this is a form of compound authentication. Loosely
+speaking, compound authentication is the property that an execution of multiple authentication
+protocols, wherein at least one is uncompromised, jointly authenticates all protocols.
```
Comment (jhoyla) [inline draft-ietf-tls-external-psk-importer.md]: "Specifically the uncompromised protocol execution proves that all the other protocols bound to it are authentic."
Should we mention that some authentication protocols only authenticate earlier protocols, and some only authenticate later ones? This fact is derivable from the reference, but is not explicitly mentioned.

File: draft-ietf-tls-external-psk-importer.md [hunk 3]
```diff
@@ -195,7 +203,33 @@ PSKs for TLS 1.2. Indeed, this is necessary for incremental deployment.
 
 # Security Considerations
 
-DISCLAIMER: This is a WIP draft and has not yet seen significant security analysis.
+The Key Importer security goals can be roughly stated as follows: avoid PSK re-use across
```
Comment (jhoyla) [inline draft-ietf-tls-external-psk-importer.md]: Is there a distinction to be made between goals of the protocol and the properties we claim it achieves?
I would say preventing IKM reuse is a goal, but not a property. The property is key separation by KDF. 
Maybe this is unnecessarily specific.

File: draft-ietf-tls-external-psk-importer.md [hunk 4]
```diff
@@ -195,7 +203,33 @@ PSKs for TLS 1.2. Indeed, this is necessary for incremental deployment.
...
+KDFs.
+
+PSK-based authentication is functionally equivalent to session resumption in that a connection
+uses existing key material to authenticate both endpoints. Following the work of
+{{?BAA15=DOI.10.14722/ndss.2015.23277}}, this is a form of compound authentication. Loosely
+speaking, compound authentication is the property that an execution of multiple authentication
+protocols, wherein at least one is uncompromised, jointly authenticates all protocols.
+Authenticating with an externally provisioned PSK, therefore, should ideally authenticate both
+the TLS connection and the external provision process. Typically, the external provision process
+produces a PSK and corresponding context in which the PSK should be used. We refer to an external
```
Comment (jhoyla) [inline draft-ietf-tls-external-psk-importer.md]: The context is more about where the key came from than where it is going. 
"corresponding context about how the PSK was derived."

Comment (chris-wood) [reply draft-ietf-tls-external-psk-importer.md]: Hmm... we use `context` for selfie mitigations, which describe where it's going, right? It seems like to can go both ways.

File: draft-ietf-tls-external-psk-importer.md (hunk 3 above)
Comment (chris-wood) [reply draft-ietf-tls-external-psk-importer.md]: It's probably best to keep them separate, yeah. Good suggestion! I'll try and clear this up.

File: draft-ietf-tls-external-psk-importer.md (hunk 2 above)
Comment (chris-wood) [reply draft-ietf-tls-external-psk-importer.md]: Hmm, is that detail relevant here? (I don't this so, but you are the expert!)

File: draft-ietf-tls-external-psk-importer.md (hunk 1 above)
Comment (chris-wood) [reply draft-ietf-tls-external-psk-importer.md]: How about step(s)?

**Pull request merged**